src/util/namespace.h

```diff
@@ -84,8 +84,14 @@ class namespacet:public namespace_baset
 
   using namespace_baset::lookup;
 
-  // these do the actual lookup
+
+  /// Searches for a symbol named \p name in `symbol_table1` and returns a
+  /// pointer to it in output parameter \p symbol when found. When no symbol is
+  /// found, it performs the same lookup in `symbol_table2`.
+  /// \return False iff the requested symbols is found in one of the two tables.
   virtual bool lookup(const irep_idt &name, const symbolt *&symbol) const;
+
+  // this and lookup above do the actual lookup
```

Comment:
Many thanks for the detailed explanation. I included a reworded version of this in the corresponding commit.